### ISSUE TITLE
Parsing response's content ranges

### ISSUE THREAD

**Issue by nyurik** [bancek/rust-http-range#8]
Hi, it seems there is no other rust code for parsing ranges, but you only handle the request header's structure, not the response (which is slightly different). Would you be ok to add that to your lib? Eventually I do hope the `reqwest` lib will support it directly (I created a [ticket](https://github.com/seanmonstar/reqwest/issues/1468) there), but noone has replied yet.

**Comment by bancek** [comment]
I'd be happy to include that in the library.

**Comment by nyurik** [comment]
@bancek thx, are you ok with the proposed implementation structure? (see reqwest ticket above)

**Comment by bancek** [comment]
I'm not that familiar with the response part of but i think it looks good. We'll see if anything feels off when you write some tests.

**Comment by nyurik** [comment]
@bancek ok, sounds good. One important difference with your implementation is that it keeps values "as is" -- first byte and last byte, rather than first + length. It keeps things less surprising for the users. Will submit something soonish.

**Comment by nyurik** [comment]
As part of my implementation I realized you may have an integer overflow bug. I think this implementation should fix it, but I didn't want to do a PR because I changed the method signature -- IMO Option<u64> is easier to use here. Plus no more hexes.  This test fails in debug mode, but passes with `caro test --release`:

```none
            T(
                "bytes=0-99999999999999999999999999999999999999999999",
                10,
                vec![HttpRange {
                    start: 0,
                    length: 10,
                }],
            ),
```

```rust
    fn parse_u64(&self) -> Option<u64> {
        if self.is_empty() {
            return None;
        }
        let mut res = 0u64;
        for b in self {
            if *b >= b'0' && *b <= b'9' {
                res = res.checked_mul(10)?.checked_add((b - b'0') as u64)?;
            } else {
                return None;
            }
        }

        Some(res)
    }
```

**Comment by nyurik** [comment]
After a lot of thought, I realized that mixing `range` and `content-range` might actually be more confusing than helpful - while similar, they serve very different usages -- server vs client.  So I used some ideas from your crate to create a new one -- [http-content-range](https://crates.io/crates/http-content-range). I have not done any benchmarking to see if my parsing approach is faster or slower, but might be a fun little project later on.  Thank you for all your wonderful feedback and the great crate code!
